### PR TITLE
Add missing error handling; fix a Docker deprecation warning

### PR DESCRIPTION
**Pull request by bradlarsen**

### CODE REVIEW

File: pkg/asm/capabilities/github.go [hunk 1]
```diff
@@ -34,6 +35,9 @@ func (task *Git) Invoke() error {
 	re := regexp.MustCompile(`^(https://)?github\.com/([^/]+)/?$`)
 	org := task.Asset.Name
 	matches := re.FindStringSubmatch(org)
+	if matches == nil {
```
Comment (privateducky) [inline pkg/asm/capabilities/github.go:38]: this should get caught by the Match function. is it not?

Comment (bradlarsen) [reply pkg/asm/capabilities/github.go:38]: Do you mean the `matches := re.FindStringSubmatch(org)` line above? That returns `nil` if the regex doesn't matches. In that case, then `org = matches[len(matches)-1]` line following ends up indexing at `matches[-1]`, giving a panic. This causes the capability to crash hard, leaving a failed job in the `running` state that hangs around until a cron job cleans it up...

Comment (privateducky) [reply pkg/asm/capabilities/github.go:38]: no; a capability will only get invoked if it passes the Match function, which exists in every capability. this function exists to ensure the target asset is the right type for the capability.

Comment (bradlarsen) [reply pkg/asm/capabilities/github.go:38]: I see. I _do_ see a panic at the `org = matches[len(matches)-1]` line in my dev backend, so somehow that `Match` checking is insufficient.

Let me take a closer look as to what's going on.

Comment (bradlarsen) [reply pkg/asm/capabilities/github.go:38]: Indeed, the `Git.Match` function doesn't check. It is this:

https://github.com/praetorian-inc/chariot-client/blob/91ac10831c66f6e2c02f21ff5acdcb233273cf27/pkg/asm/capabilities/github.go#L24-L26

...and `task.Asset.Is` is this:

https://github.com/praetorian-inc/chariot-client/blob/91ac10831c66f6e2c02f21ff5acdcb233273cf27/pkg/sdk/model/asset.go#L104-L106

No checking of the `name` is done there to ensure that it matches the regex format.

Comment (bradlarsen) [reply pkg/asm/capabilities/github.go:38]: Here's the stack trace I was seeing in my local testing:
```
time=2024-09-04T14:21:16.144Z level=INFO msg=receipt job=github asset=#asset#github#https://github.com/mozilla-appmaker username=bradford.larsen@praetorian.com
time=2024-09-04T14:21:16.541Z level=ERROR msg="job failed" composite=#job#github#https://github.com/mozilla-appmaker#github error="GET https://api.github.com/user/repos?per_page=100: 401 Bad credentials []" username=bradford.larsen@praetorian.com
time=2024-09-04T14:21:16.584Z level=INFO msg=complete job=github asset=#asset#github#https://github.com/mozilla-appmaker duration=0 username=bradford.larsen@praetorian.com
time=2024-09-04T14:21:16.746Z level=INFO msg=receipt job=github asset=#asset#github#setup username=bradford.larsen@praetorian.com
panic: runtime error: index out of range [-1]

goroutine 1 [running]:
github.com/praetorian-inc/chariot-client/pkg/asm/capabilities.(*Git).Invoke(0xc0001fefc8)
        /go/pkg/mod/github.com/praetorian-inc/chariot-client@v0.0.0-20240903102941-11e8ffc14146/pkg/asm/capabilities/github.go:37 +0x525
chariot/internal/compute.Process.func8()
        /app/internal/compute/registry.go:194 +0x62
chariot/internal/compute.Process({{0xc0002b1340, 0x1e}, {0xc0001f0570, 0x18}, {0xc000410a08, 0x6}, {0xc000410a20, 0x6}, {0x0, 0x0}, ...})
        /app/internal/compute/registry.go:200 +0x478
main.process({0xcd9f60, 0x122d8e0}, 0xc0003f2000, {0xc0000405b7, 0x41})
        /app/cmd/listeners/compute/main.go:88 +0x618
main.main()
        /app/cmd/listeners/compute/main.go:140 +0x146
make: *** [compute] Error 2
```

This was when attempting to scan an Attack Surface I added for `https://github.com/mozilla-appmaker`, but without providing a PAT. Connecting the integration succeeded, but then the scanning jobs fail due to invalid credentials, leading to the downstream panic I'm patching around in this PR.

There is some broken error handling elsewhere in this capability. I'll push another commit...

Comment (privateducky) [reply pkg/asm/capabilities/github.go:38]: awesome context.

any reason not to update the ``Match`` function with better matching/regex logic?

also: a separate, more generic commit went in last week (by @praetorian-rad) which ensures we're checking the ``Match`` function when running capabilities locally (we previously were not). I'm curious if this error can be reproduced after that change.

**Review by privateducky — CHANGES_REQUESTED** [review]
Not sure a) if this is still valid and b) if this logic is in the right spot

**Comment by bradlarsen** [comment]
> Not sure a) if this is still valid and b) if this logic is in the right spot

(a) I just checked the latest commit of chariot-client, and the relevant code hasn't changed. I think this is still valid.

(b) Where should this logic go, if not where I've put it (at the point where the code can go wrong)?

You asked earlier if the `Match` function catches this. [It does not](https://github.com/praetorian-inc/chariot-client/pull/28#discussion_r1744084459). The `Match` function does some validity check, but it's insufficient for preventing the crash fixed in this PR. The actual state that the `Invoke` function does its regex matching on is populated somewhere in between the call to `Match` and `Invoke`.

File: pkg/asm/capabilities/github.go (hunk 1 above)
Comment (bradlarsen) [reply pkg/asm/capabilities/github.go]: Whoops, I missed this comment initially! I've updated `Match` to move the validity checking earlier now.

**Review by privateducky — APPROVED** [review]

**Pull request merged**